### PR TITLE
NAS-132812 / 25.04 / Convert listdir and mkdir to api_method

### PR DESCRIPTION
**Pull request by anodos325**
This commit converts the two filesystem API endpoints `filesystem.listdir` and `filesystem.mkdir` to new api_method.

As part of this change, the deprecated way of calling filesystem.mkdir using positional arguments was removed.

### CODE REVIEW

File: src/middlewared/middlewared/api/v25_04_0/filesystem.py [hunk 1]
```diff
@@ -79,3 +86,97 @@ def payload_is_actionable(self) -> Self:
...
+    """ Canonical path of the entry, eliminating any symbolic links"""
+    type: Literal[
+        StatxEtype.DIRECTORY,
+        StatxEtype.FILE,
+        StatxEtype.SYMLINK,
+        StatxEtype.OTHER,
+    ]
+    size: int
+    """ Size in bytes of a plain file. """
+    allocation_size: int
```
Comment (yocalebo) [inline src/middlewared/middlewared/api/v25_04_0/filesystem.py:136]: Missing docstring `allocation_size`?

File: src/middlewared/middlewared/api/v25_04_0/filesystem.py [hunk 2]
```diff
@@ -79,3 +86,97 @@ def payload_is_actionable(self) -> Self:
...
+    xattrs: list[NonEmptyString]
+    """ List of xattr names of extended attributes on file. """
+    zfs_attrs: list[FILESYSTEM_ZFS_ATTRS] | None
+    """ List of extra ZFS-related file attribute indicators on file. Will be None type if filesystem is not ZFS. """
+
+
+class FilesystemListdirArgs(BaseModel):
+    path: NonEmptyString
+    query_filters: QueryFilters = []
+    query_options: QueryOptions = QueryOptions()
```
Comment (yocalebo) [inline src/middlewared/middlewared/api/v25_04_0/filesystem.py:164]: @creatorcary I don't know if doing `QueryOptions` is the "correct" way of populating this field with default values? Do we need to do something like `Field(default_factory=QueryOptions`?

Comment (anodos325) [reply src/middlewared/middlewared/api/v25_04_0/filesystem.py:165]: This is how default is set in `QueryArgs` in common.py.

Comment (creatorcary) [reply src/middlewared/middlewared/api/v25_04_0/filesystem.py:165]: Pretty sure either way works.

File: tests/api2/test_435_smb_registry.py [hunk 3]
```diff
@@ -70,7 +70,7 @@ def create_smb_share(path, share_name, mkdir=False, options=None):
     cr_opts = options or {}
 
     if mkdir:
-        call('filesystem.mkdir', path)
+        call('filesystem.mkdir', {'path': path, 'options': {'raise_chmod_error':   False}})
```
Comment (mgrimesix) [inline tests/api2/test_435_smb_registry.py:73]: Flake8: too many spaces before `False`
Same with the other two below.

Also... 
Why is it now necessary to pass `{'raise_chmod_error': False}`  ?

Comment (anodos325) [reply tests/api2/test_435_smb_registry.py:73]: Because SMB datasets have a restricted ZFS aclmode.

**Review by mgrimesix — APPROVED** [review]

**Pull request merged**